crates/egui_demo_lib/src/demo/modals.rs

```diff
@@ -0,0 +1,250 @@
+use egui::{Align, ComboBox, Context, Id, Layout, Modal, ProgressBar, Ui, Widget, Window};
+
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(default))]
+pub struct Modals {
+    user_modal_open: bool,
+    save_modal_open: bool,
+    save_progress: Option<f32>,
+
+    role: &'static str,
+    name: String,
+}
+
+impl Default for Modals {
+    fn default() -> Self {
+        Self {
+            user_modal_open: false,
+            save_modal_open: false,
+            save_progress: None,
+            role: Self::ROLES[0],
+            name: "John Doe".to_owned(),
+        }
+    }
+}
+
+impl Modals {
+    const ROLES: [&'static str; 2] = ["user", "admin"];
+}
+
+impl crate::Demo for Modals {
+    fn name(&self) -> &'static str {
+        "🗖 Modals"
+    }
+
+    fn show(&mut self, ctx: &Context, open: &mut bool) {
+        use crate::View as _;
+        Window::new(self.name())
+            .open(open)
+            .vscroll(false)
+            .resizable(false)
+            .show(ctx, |ui| self.ui(ui));
+    }
+}
+
+impl crate::View for Modals {
+    fn ui(&mut self, ui: &mut Ui) {
+        let Self {
+            user_modal_open,
+            save_modal_open,
+            save_progress,
+            role,
+            name,
+        } = self;
+
+        ui.horizontal(|ui| {
+            if ui.button("Open User Modal").clicked() {
+                *user_modal_open = true;
+            }
+
+            if ui.button("Open Save Modal").clicked() {
+                *save_modal_open = true;
+            }
+        });
+
+        ui.label("Click one of the buttons to open a modal.");
+        ui.label("Modals have a backdrop and prevent interaction with the rest of the UI.");
+        ui.label(
+            "You can show modals on top of each other and close the top most modal with \
+            escape or by clicking outside the modal.",
+        );
+
+        if *user_modal_open {
+            let modal = Modal::new(Id::new("Modal A")).show(ui.ctx(), |ui| {
+                ui.set_width(250.0);
+
+                ui.heading("Edit User");
+
+                ui.label("Name:");
+                ui.text_edit_singleline(name);
+
+                ComboBox::new("role", "Role")
+                    .selected_text(*role)
+                    .show_ui(ui, |ui| {
+                        for r in Self::ROLES {
+                            ui.selectable_value(role, r, r);
+                        }
+                    });
+
+                ui.separator();
+
+                ui.with_layout(Layout::right_to_left(Align::Min), |ui| {
+                    if ui.button("Save").clicked() {
+                        *save_modal_open = true;
+                    }
+                    if ui.button("Cancel").clicked() {
+                        *user_modal_open = false;
+                    }
+                });
```

Comment:
We could use `egui::Sides` here (and below) to have the buttons on the left/right edges